### PR TITLE
Load custom jar in JMX command

### PR DESCRIPTION
**Pull request by therve**
### What does this PR do?

It loads `custom_jar_paths` from integrations `init_config` calling `agent jmx` commands.

### Motivation

I need to test jboss integration, which requires a custom jar, and currently it
fails to run because it doesn't add the jar to the classpath.

### Additional Notes

I haven't tested it yet.

### CODE REVIEW

**Review by olivielpeau — CHANGES_REQUESTED** [review]
Thanks for looking into this! A few comments:

* I think the logic you added to `cmd/agent/app/jmx.go` should actually be extracted out from the runner code (https://github.com/DataDog/datadog-agent/blob/340a6860b4204cc1e9d52ae937807252b93a59f8/pkg/collector/corechecks/embed/jmx/runner.go#L76-L107) to a method on the `JMXFetch` struct, so that both code paths can simply call that method. What do you think?
* the unit tests are failing on CircleCI, on what look like legitimate errors
* can you add a release note? (docs here: https://github.com/DataDog/datadog-agent/blob/master/docs/dev/contributing.md#reno)

(let us know if you don't have time to address this, and you'd like to hand off the next steps to Agent Core 😃  )

**Comment by therve** [comment]
Thanks, I think I handled everything, let me know how it looks.

File: releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
```
Comment (olivielpeau) [inline releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml]: this is really a bugfix IMO, so should be under `fixes`:

```suggestion
fixes:
```

File: releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - |
+    We now load JMX related options from integration configurations when
+    calling the `agent jmx` commands.
```
Comment (olivielpeau) [inline releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml]: I'd change this to:

```
The `agent jmx` command now correctly takes into account the options in the `init_config` section of the JMXFetch integration configs
```

File: pkg/collector/check/jmx.go [hunk 3]
```diff
@@ -13,16 +13,18 @@ import (
 )
 
 // IsJMXConfig checks if a certain YAML config is a JMX config
-func IsJMXConfig(name string, initConf integration.Data) bool {
+func IsJMXConfig(name string, initConf integration.Data, rawInitConfig integration.RawMap) bool {
```
Comment (olivielpeau) [inline pkg/collector/check/jmx.go]: I don't understand why this change is still necessary 🤔

**Review by olivielpeau — COMMENTED** [review]
A couple of additional comments, overall LGTM 👍 

Were you able to actually test these changes?

File: releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml (hunk 1 above)
Comment (therve) [reply releasenotes/notes/cmd-jmx-options-6a1a75d4bbf62ca2.yaml]: Done

File: cmd/agent/app/jmx.go [hunk 4]
```diff
@@ -191,6 +191,7 @@ func loadConfigs() {
 		if check.IsJMXConfig(c.Name, c.InitConfig) && (includeEverything || configIncluded(c)) {
 			fmt.Println("Config ", c.Name, " was loaded.")
 			jmx.AddScheduledConfig(c)
+			runner.ConfigureCheck(c.InitConfig)
```
Comment (olivielpeau) [inline cmd/agent/app/jmx.go]: any reason you only call `ConfigureCheck` here instead of both `ConfigureCheck` and `ConfigureInstance`?

File: pkg/jmxfetch/jmxfetch.go [hunk 5]
```diff
@@ -267,3 +285,71 @@ func (j *JMXFetch) Up() (bool, error) {
 	err = process.Signal(syscall.Signal(0))
 	return err == nil, err
 }
+
+func (j *JMXFetch) ConfigureCheck(initConfig integration.Data) error {
```
Comment (olivielpeau) [inline pkg/jmxfetch/jmxfetch.go]: this function is exported, so the go linter should require a small docstring on top of it, not sure why it's not catching it. Can you add one?

Also, I'd prefer a name like `ConfigureFromInitConfig` to reflect that we're configuring the `JMXFetch` struct from what's in `initConfig`, which sounds more explicit to me than `ConfigureCheck`. WDYT?

File: pkg/jmxfetch/jmxfetch.go [hunk 6]
```diff
@@ -267,3 +285,71 @@ func (j *JMXFetch) Up() (bool, error) {
...
+	if j.JavaCustomJarPaths == nil {
+		if initConf.CustomJarPaths != nil {
+			j.JavaCustomJarPaths = initConf.CustomJarPaths
+		}
+	}
+
+	return nil
+}
+
+func (j *JMXFetch) ConfigureInstance(instance integration.Data) error {
```
Comment (olivielpeau) [inline pkg/jmxfetch/jmxfetch.go]: ditto for the doc string, and in terms of naming I'd prefer `ConfigureFromInstance`

**Review by olivielpeau — COMMENTED** [review]
A few more comments, and I'll ask again: were you actually able to test the changes? :)

File: pkg/jmxfetch/jmxfetch.go (hunk 5 above)
Comment (therve) [reply pkg/jmxfetch/jmxfetch.go]: Good idea, changed.

File: pkg/jmxfetch/jmxfetch.go (hunk 6 above)
Comment (therve) [reply pkg/jmxfetch/jmxfetch.go]: Done.

File: cmd/agent/app/jmx.go (hunk 4 above)
Comment (therve) [reply cmd/agent/app/jmx.go]: Wasn't sure it was worth iterating on instances, done now.

**Comment by therve** [comment]
> A few more comments, and I'll ask again: were you actually able to test the changes? :)

No, I wasn't able to. Is there a build available in CI to test?

File: pkg/collector/check/jmx.go (hunk 3 above)
Comment (therve) [reply pkg/collector/check/jmx.go]: Removed...

**Review by olivielpeau — APPROVED** [review]
LGTM once you've confirmed it works well after testing!

You'll find artifacts for Linux and Windows on Gitlab, in the pipeline on your branch, in the related job of the `package_build` step

**Comment by therve** [comment]
Tested, it works great. Thanks!
